shared/cancel/canceller.go

```diff
@@ -0,0 +1,19 @@
+package cancel
+
+import "context"
+
+// Canceller is a simple wrapper for a cancellable context which makes the associated context.CancelFunc more easily
+// accessible.
+type Canceller struct {
+	context.Context
+	Cancel func()
```

Comment:
@markylaing oh can we make the type of this field `context.CancelFunc` for clarity.